DemoServer/Controllers/Demos/JavascriptIndexes/JavascriptMapIndex/metadata.json

```diff
@@ -0,0 +1,161 @@
+                }
+            ]
+        },
+        {
+            "Title": "The index constructor",
+            "Slug": "step-3",
+            "DescriptionLines": [
+                "* Define `Maps` which is a _HashSet_ that will contain a single javascript map strings (Map index), or multiple map strings (Multi Map index).",
+                "",
+                "* The map string is defined with: `map(<Collection name>, <JavaScript map method>)`.",
+                "",
+                "* The _JavaScript map method_ returns a JavaScript object literal.<br>",
+                "  The properties of this object are the fields composing the index entry.",
+                "",
+                "* Any logic written in JavaScript can be applied to the properties values, e.g.:",
+                "  * FullName - combining 2 fields ('FirstName' & 'LastName')",
+                "  * Country - indexing nested field data ('Country' is nested under 'Address')",
+                "  * WorkingInCompanySince - indexing partial field data (only the 'Year')",
+                "  * NumberOfTerritories - applying logic (using length)",
+                "",
+                "* The index built will contain a list of _indexed terms_ for each _index-field_ specified in the map function, and a mapping to the relevant document.",
+                "  i.e. 'USA' & 'UK' are terms that will be indexed under index-field 'Country', along with the relevant matching documents IDs.",
+                "",
+                "* These indexed terms can later be queried on to get their matching documents, the documents that contain them."
+            ],
+            "Assets": [
+                {
+                    "Title": "Map Indexes",
+                    "Type" : "link",
+                    "Url"  : "https://ravendb.net/docs/article-page/latest/csharp/indexes/map-indexes"
+                },
+                {
+                    "Title": "JavaScript indexes",
+                    "Type" : "link",
+                    "Url"  : "https://ravendb.net/docs/article-page/latest/csharp/indexes/javascript-indexes"
+                }
+            ]
+        },
+        {
+            "Title": "Query for documents",
+            "Slug" : "step-4",
+            "DescriptionLines": [
+                "* Query the 'Employees' collection using the defined index.",
+                "",
+                "* The first generic _Employees_ByImportantDetails.IndexEntry_ allows the compiler to auto-complete the index entry fields on which you want to query in the .Where clause.",
+                "",
+                "* _RavenDB will search the index_ for the indexed terms that match the query criteria", 
+                "  and retrieve the matching documents (according to the document ID listed in the index) from the documents store.",
+                "",
+                "* The query results will include all documents from the 'Employees' collection whose 'Country' is 'USA',",
```

Comment:
Just to check - is country intentionally hardcoded or do we want to introduce it as a parameter on top of the page and read it from there?

Reply:
At the time, when I did the '**map index**' demo, I chose to hardcode 'USA' and have only one param.
Now, when doing the '**js map index**' demo, I thought it should be a twin demo to the 'map index' one
so that one can easily spot the difference with the map definition.

Do you think it is necessary to change the country to be a parameter ?
In that case, it should be done for both demos.

Reply:
No, not necessary, let's go with this version